src/full/Agda/Syntax/Parser/Parser.y

```diff
@@ -721,7 +723,8 @@ Expr3NoCurly
     | setN                              { SetN (getRange (fst $1)) (snd $1) }
     | propN                             { PropN (getRange (fst $1)) (snd $1) }
     | '{{' Expr DoubleCloseBrace        { InstanceArg (getRange ($1,$2,$3)) (maybeNamed $2) }
-    | '(|' Expr '|)'                    { IdiomBrackets (getRange ($1,$2,$3)) $2 }
+    | '(|' WithExprs '|)'               { IdiomBrackets (getRange ($1,$2,$3)) $2 }
+    | '(|)'                             { IdiomBrackets (getRange $1) [] }
```

Comment:
I am a bit uneasy about reusing `WithExprs` given that e.g. I don't expect to be able to write dotted expressions in an idiom bracket.

Reply:
I think `WithExprs` is the thing we write on the first line when specifying the expressions to do `with` on, so there are no dot patterns allowed.

Reply:
My bad.